lib/models.js

```diff
@@ -1182,6 +1182,46 @@ class BooleanConstraint extends Constraint {
   }
 }
 
+// FixedValueConstraint only makes sense when fixing a value
+class FixedValueConstraint extends Constraint {
+
+  constructor(value, type, path) {
+    super(path);
+    this._value = value;
+    this._type = type;
+  }
+
+  get value() { return this._value; }
+  get type() { return this._type; }
+
+  clone() {
+    const clone = new FixedValueConstraint(this._value, this._type);
+    this._clonePropertiesTo(clone);
+    return clone;
+  }
+
+  equals(other) {
+    return (other instanceof FixedValueConstraint) &&
+        this._value == other.value &&
+        this._type == other.type &&
+        this._pathsAreEqual(other);
+  }
+
+  toJSON() {
+    var constraint = super.toJSON();
+    delete constraint['path'];
+
+    return Object.assign({
+      'type': this.type,
+      'value': this.value,
+    }, constraint);
```

Comment:
I wonder if it might be best to identify the `type` as `fixed-value` and add a `valueType` property of `string`?